src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl

```diff
@@ -1,207 +1,247 @@
-    rbotMin::T, rbotMax::T, rtopMin::T, rtopMax::T = get_r_limits(c)
-    zMin::T, zMax::T = get_z_limits(c)
-    φMin::T, φMax::T, _ = get_φ_limits(c)
-    surfaces = _get_decomposed_surfaces_cone(c, rbotMin, rbotMax, rtopMin, rtopMax, zMin, zMax)
-    if !_is_cone_collapsed(rbotMin, rbotMax, rtopMin, rtopMax, zMin, zMax)
-        push!(surfaces, ConalPlane(c, φ = φMin), ConalPlane(c, φ = φMax))
-        push!(surfaces, ConeMantle(c, rbot = rbotMin, rtop = rtopMin))
-    end
-    surfaces
-end
+# in(p::AbstractCoordinatePoint, c::Cone{<:Any, <:Any, <:AbstractInterval, <:Any}) =
+#     _in_z(p, c.z) && _in_φ(p, c.φ) && _in_cyl_r(p, get_r_at_z(c, p.z))
 
-function sample(c::Cone{T}, step::Real)::Vector{CylindricalPoint{T}} where {T}
-    zMin::T, zMax::T = get_z_limits(c)
-    φMin::T, φMax::T, _ = get_φ_limits(c)
-    samples = [
-        CylindricalPoint{T}(r,φ,z)
-        for z in zMin:step:zMax
-        for r in _left_radial_interval(get_r_at_z(c, z)):step:_right_radial_interval(get_r_at_z(c, z))
-        for φ in (r == 0 ? φMin : φMin:step/r:φMax)
-    ]
-end
-
-function sample(c::Cone{T}, Nsamps::NTuple{3,Int})::Vector{CylindricalPoint{T}} where {T}
-    zMin::T, zMax::T = get_z_limits(c)
-    φMin::T, φMax::T, _ = get_φ_limits(c)
-    samples = [
-        CylindricalPoint{T}(r,φ,z)
-        for z in (Nsamps[3] ≤ 1 ? zMin : range(zMin, zMax, length = Nsamps[3]))
-        for r in (Nsamps[1] ≤ 1 ? _left_radial_interval(get_r_at_z(c, z)) : range(_left_radial_interval(get_r_at_z(c, z)), _right_radial_interval(get_r_at_z(c, z)), length = Nsamps[1]))
-        for φ in (Nsamps[2] ≤ 1 ? φMin : range(φMin, φMax, length = Nsamps[2]))
-    ]
-end
-
-@inline sample(c::Cone{T}) where {T} = sample(c, (2,3,3))
-@inline sample(c::Cone{T, <:Any, Nothing}) where {T} = sample(c, (2,5,3))
+# # read-in
+function Geometry(::Type{T}, t::Type{Cone}, dict::AbstractDict, input_units::NamedTuple, transformations::Transformations{T}) where {T}
+    length_unit = input_units.length
+    angle_unit = input_units.angle
+    r = parse_r_of_primitive(T, dict, length_unit)
+    φ = parse_φ_of_primitive(T, dict, angle_unit)
+    hZ = parse_height_of_primitive(T, dict, length_unit)
+    cone = Cone{T, ClosedPrimitive, typeof(r), typeof(φ)}(r = r, φ = φ, hZ = hZ)
+    return transform(cone, transformations)
+end
+
+function surfaces(t::Tube{T}) where {T}
+    bot_center_pt = _transform_into_global_coordinate_system(CartesianPoint{T}(zero(T), zero(T), -t.hZ), t) 
+    top_center_pt = _transform_into_global_coordinate_system(CartesianPoint{T}(zero(T), zero(T), +t.hZ), t) 
+    mantle = ConeMantle{T,T,Nothing}(t.r, t.φ, t.hZ, t.origin, t.rotation)
+    bot_normal = top_center_pt - bot_center_pt
+    e_bot = EllipticalSurface{T,T,Nothing}(r = t.r, φ = nothing, origin = bot_center_pt, rotation =  t.rotation)
+    e_top = EllipticalSurface{T,T,Nothing}(r = t.r, φ = nothing, origin = top_center_pt, rotation = -t.rotation)
```

Comment:
Why not `RotZ(π) * -t.rotation` here?

Reply:
Implemented at different times. 
I have to think about it. It might be that the `φ` has to be also changed if this not full in φ.